frontend/src/pages/pipelines/global/experiments/artifacts/ArtifactsTable.tsx

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+import { Flex, FlexItem, TextInput, Truncate } from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { TableVariant, Td, Tr } from '@patternfly/react-table';
+
+import { Artifact as MlmdArtifact } from '~/third_party/mlmd';
+import { TableBase } from '~/components/table';
+import DashboardEmptyTableView from '~/concepts/dashboard/DashboardEmptyTableView';
+import PipelinesTableRowTime from '~/concepts/pipelines/content/tables/PipelinesTableRowTime';
+import { FilterToolbar } from '~/concepts/pipelines/content/tables/PipelineFilterBar';
+import SimpleDropdownSelect from '~/components/SimpleDropdownSelect';
+import { ArtifactType } from '~/concepts/pipelines/kfTypes';
+import { useMlmdListContext } from '~/concepts/pipelines/context';
+import { FilterOptions, columns, initialFilterData, options } from './constants';
+
+interface ArtifactsTableProps {
+  artifacts: MlmdArtifact.AsObject[] | null | undefined;
+  nextPageToken: string | undefined;
+  isLoaded: boolean;
+}
+
+export const ArtifactsTable: React.FC<ArtifactsTableProps> = ({
+  artifacts,
+  isLoaded,
+  nextPageToken,
+}) => {
+  const {
+    maxResultSize,
+    setFilterQuery,
+    setPageToken: setRequestToken,
+    setMaxResultSize,
+  } = useMlmdListContext();
+  const [page, setPage] = React.useState(1);
+  const [filterData, setFilterData] = React.useState(initialFilterData);
+  const onClearFilters = React.useCallback(() => setFilterData(initialFilterData), []);
+  const [pageTokens, setPageTokens] = React.useState<Record<number, string>>({});
+
+  // Force disabled state to pagination when there is no nextPageToken
+  React.useEffect(() => {
+    const paginationNextButtons = document.querySelectorAll('button[aria-label="Go to next page"]');
+
+    paginationNextButtons.forEach((button) => {
+      if (!nextPageToken) {
+        button.setAttribute('disabled', '');
+      } else {
+        button.removeAttribute('disabled');
+      }
+    });
+  }, [nextPageToken]);
```

Comment:
Can you make this a hook file like `useDisablePaginationNextButtons(nextPageToken)`? It would be easier to reuse it in the execution table as well.

Reply:
I don't think that's necessary and that's a confusing name, where its also only applicable to MLMD list tables.

Reply:
I moved the logic inside of `useMlmdListContext` so you get it for free with executions.